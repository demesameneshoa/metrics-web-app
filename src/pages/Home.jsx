import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Header from '../components/Header';
import Hero from '../components/Hero';
import Country from '../components/Countries';
import { fetchAirQualityC } from '../redux/AQSlice';

const Home = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const countries = useSelector((state) => state.airQuality.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    countries.forEach((country) => {
      dispatch(fetchAirQualityC(country.name));
    });
  }, [dispatch, countries]);

  const filteredCountries = countries.filter(
    (country) => country.name.toLowerCase().includes(searchTerm.toLowerCase()),
  );

  return (
    <>
      <div className="App">
        <Header previous=" " />
        <Hero title="" searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
        <div className="countries">
          {filteredCountries.map((country, index) => (
            <Country
              key={country.name}
              name={country.name}
              vector={country.map}
              cities={country.cities}
              airquality={country.airquality}
              isOdd={index % 2 !== 0}
            />
          ))}
        </div>
      </div>
    </>
  );
};

export default Home;
