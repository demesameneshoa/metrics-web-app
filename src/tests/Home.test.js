import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Home from '../pages/Home';
import { BrowserRouter } from 'react-router-dom';

describe('Home unit test', () => {
  describe('Home page Renders Properly ', () => {
    const component = (
        <Provider store={configureStore}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
     </Provider>
      );
      test('Home Page component renders properly ', () => {
        const tree = render(component);
        expect(tree).toMatchSnapshot();
      });
      test('Should render the correct hero', () => {
        render(component);
        const element = screen.getByText('Air Quality Metrics');
        expect(element).toBeInTheDocument();
      });
})
});