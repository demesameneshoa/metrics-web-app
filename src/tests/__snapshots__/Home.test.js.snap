// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Renders Properly  Home Page component renders properly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="header-container"
        >
          <a
            href="/ "
            style="text-decoration: none; color: white;"
          />
          <div
            class="header-icons"
          >
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-microphone "
                data-icon="microphone"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M192 0C139 0 96 43 96 96V256c0 53 43 96 96 96s96-43 96-96V96c0-53-43-96-96-96zM64 216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 70.7-57.3 128-128 128s-128-57.3-128-128V216z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-gear "
                data-icon="gear"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="hero"
        >
          <h2>
            Air Quality Metrics
            <div>
              <input
                class="seaarchbar"
                placeholder="Search..."
                type="text"
                value=""
              />
            </div>
          </h2>
        </div>
        <div
          class="countries"
        >
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-bcXGWu cDCZPf country"
              vector="Canada.png"
            >
              <h3>
                Canada
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-gswPWN ccDIJW country"
              vector="Jamica.png"
            >
              <h3>
                Jamaica
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-dkrGVk eQhDEN country"
              vector="Mexico.png"
            >
              <h3>
                Mexico
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hLBbCX kyEBa-d country"
              vector="USA.png"
            >
              <h3>
                USA
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-eDvSBu hrTLKd country"
              vector="Argentina.png"
            >
              <h3>
                Argentina
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-jSVbBN ketkQC country"
              vector="Brazil.png"
            >
              <h3>
                Brazil
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-gKPSAk gbXVBh country"
              vector="Ecuador.png"
            >
              <h3>
                Ecuador
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-iBYQGD kCIlvF country"
              vector="Uruguay.png"
            >
              <h3>
                Uruguay
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-ftTHFa gJAlVn country"
              vector="Belgium.png"
            >
              <h3>
                Belgium
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-pyfiu bqfbJw country"
              vector="Denmark.png"
            >
              <h3>
                Denmark
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-jrcSnH BpIuQ country"
              vector="Italy.png"
            >
              <h3>
                Italy
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-kDvvrc jXWqrF country"
              vector="Portugal.png"
            >
              <h3>
                Portugal
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-ipEyXt hHhZfo country"
              vector="Cameroon.png"
            >
              <h3>
                Cameroon
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-csuROW fEyqHp country"
              vector="Ethiopia.png"
            >
              <h3>
                Ethiopia
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-eDVvn YFtwc country"
              vector="Kenya.png"
            >
              <h3>
                Kenya
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-bqWyyI fNIgFV country"
              vector="Tunisia.png"
            >
              <h3>
                Tunisia
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-ksBlDV jyahCu country"
              vector="China.png"
            >
              <h3>
                China
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hBxdNW iyDmFJ country"
              vector="India.png"
            >
              <h3>
                India
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-fnGhun bBarqG country"
              vector="UAE.png"
            >
              <h3>
                United Arab Emirates
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-fEXoiN LzKKs country"
              vector="Yemen.png"
            >
              <h3>
                Yemen
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-bjfGFA gEGswF country"
              vector="Australia.png"
            >
              <h3>
                Australia
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-idXgxz efzJQF country"
              vector="Fiji.png"
            >
              <h3>
                Fiji
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-dIfyUm kVtKjE country"
              vector="New Zealand.png"
            >
              <h3>
                New Zealand
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
          <a
            class="country-wrapper"
            href="/cities"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hHUaPp dyeMeE country"
              vector="Samoa.png"
            >
              <h3>
                Samoa
              </h3>
              <span>
                Overall AQI:
                Loading...
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="header-container"
      >
        <a
          href="/ "
          style="text-decoration: none; color: white;"
        />
        <div
          class="header-icons"
        >
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-microphone "
              data-icon="microphone"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M192 0C139 0 96 43 96 96V256c0 53 43 96 96 96s96-43 96-96V96c0-53-43-96-96-96zM64 216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 70.7-57.3 128-128 128s-128-57.3-128-128V216z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-gear "
              data-icon="gear"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="hero"
      >
        <h2>
          Air Quality Metrics
          <div>
            <input
              class="seaarchbar"
              placeholder="Search..."
              type="text"
              value=""
            />
          </div>
        </h2>
      </div>
      <div
        class="countries"
      >
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-bcXGWu cDCZPf country"
            vector="Canada.png"
          >
            <h3>
              Canada
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-gswPWN ccDIJW country"
            vector="Jamica.png"
          >
            <h3>
              Jamaica
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-dkrGVk eQhDEN country"
            vector="Mexico.png"
          >
            <h3>
              Mexico
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-hLBbCX kyEBa-d country"
            vector="USA.png"
          >
            <h3>
              USA
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-eDvSBu hrTLKd country"
            vector="Argentina.png"
          >
            <h3>
              Argentina
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-jSVbBN ketkQC country"
            vector="Brazil.png"
          >
            <h3>
              Brazil
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-gKPSAk gbXVBh country"
            vector="Ecuador.png"
          >
            <h3>
              Ecuador
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-iBYQGD kCIlvF country"
            vector="Uruguay.png"
          >
            <h3>
              Uruguay
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-ftTHFa gJAlVn country"
            vector="Belgium.png"
          >
            <h3>
              Belgium
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-pyfiu bqfbJw country"
            vector="Denmark.png"
          >
            <h3>
              Denmark
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-jrcSnH BpIuQ country"
            vector="Italy.png"
          >
            <h3>
              Italy
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-kDvvrc jXWqrF country"
            vector="Portugal.png"
          >
            <h3>
              Portugal
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-ipEyXt hHhZfo country"
            vector="Cameroon.png"
          >
            <h3>
              Cameroon
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-csuROW fEyqHp country"
            vector="Ethiopia.png"
          >
            <h3>
              Ethiopia
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-eDVvn YFtwc country"
            vector="Kenya.png"
          >
            <h3>
              Kenya
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-bqWyyI fNIgFV country"
            vector="Tunisia.png"
          >
            <h3>
              Tunisia
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-ksBlDV jyahCu country"
            vector="China.png"
          >
            <h3>
              China
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-hBxdNW iyDmFJ country"
            vector="India.png"
          >
            <h3>
              India
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-fnGhun bBarqG country"
            vector="UAE.png"
          >
            <h3>
              United Arab Emirates
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-fEXoiN LzKKs country"
            vector="Yemen.png"
          >
            <h3>
              Yemen
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-bjfGFA gEGswF country"
            vector="Australia.png"
          >
            <h3>
              Australia
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-idXgxz efzJQF country"
            vector="Fiji.png"
          >
            <h3>
              Fiji
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-dIfyUm kVtKjE country"
            vector="New Zealand.png"
          >
            <h3>
              New Zealand
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
        <a
          class="country-wrapper"
          href="/cities"
          style="text-decoration: none; color: white;"
        >
          <div
            class="sc-hHUaPp dyeMeE country"
            vector="Samoa.png"
          >
            <h3>
              Samoa
            </h3>
            <span>
              Overall AQI:
              Loading...
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
